src/lib/block/kuznyechik/kuznyechik.cpp

```diff
@@ -4162,80 +4162,60 @@ const uint64_t C[32][2] = {{0xb87a486c7276a26e, 0x019484dd10bd275d}, {0xb3f490d8
-        T[12][get_byte<3>(x2)][1] ^ T[13][get_byte<2>(x2)][1] ^ T[14][get_byte<1>(x2)][1] ^ T[15][get_byte<0>(x2)][1];
-}
+inline void LS(uint64_t& x1, uint64_t& x2) {
+   uint64_t t1 = 0;
+   uint64_t t2 = 0;
+   for(size_t i = 0; i != 16; ++i) {
+      const uint8_t x = get_byte_var(7 - (i % 8), (i < 8) ? x1 : x2);
+      t1 ^= T[i][x][0];
+      t2 ^= T[i][x][1];
+   }
 
-inline void ILS(uint64_t x1, uint64_t x2, uint64_t& t1, uint64_t& t2) {
-   t1 = IT[0][get_byte<7>(x1)][0] ^ IT[1][get_byte<6>(x1)][0] ^ IT[2][get_byte<5>(x1)][0] ^ IT[3][get_byte<4>(x1)][0] ^
-        IT[4][get_byte<3>(x1)][0] ^ IT[5][get_byte<2>(x1)][0] ^ IT[6][get_byte<1>(x1)][0] ^ IT[7][get_byte<0>(x1)][0] ^
-        IT[8][get_byte<7>(x2)][0] ^ IT[9][get_byte<6>(x2)][0] ^ IT[10][get_byte<5>(x2)][0] ^
-        IT[11][get_byte<4>(x2)][0] ^ IT[12][get_byte<3>(x2)][0] ^ IT[13][get_byte<2>(x2)][0] ^
-        IT[14][get_byte<1>(x2)][0] ^ IT[15][get_byte<0>(x2)][0];
-   t2 = IT[0][get_byte<7>(x1)][1] ^ IT[1][get_byte<6>(x1)][1] ^ IT[2][get_byte<5>(x1)][1] ^ IT[3][get_byte<4>(x1)][1] ^
-        IT[4][get_byte<3>(x1)][1] ^ IT[5][get_byte<2>(x1)][1] ^ IT[6][get_byte<1>(x1)][1] ^ IT[7][get_byte<0>(x1)][1] ^
-        IT[8][get_byte<7>(x2)][1] ^ IT[9][get_byte<6>(x2)][1] ^ IT[10][get_byte<5>(x2)][1] ^
-        IT[11][get_byte<4>(x2)][1] ^ IT[12][get_byte<3>(x2)][1] ^ IT[13][get_byte<2>(x2)][1] ^
-        IT[14][get_byte<1>(x2)][1] ^ IT[15][get_byte<0>(x2)][1];
+   x1 = t1;
+   x2 = t2;
 }
 
-inline void ILSS(uint64_t x1, uint64_t x2, uint64_t& t1, uint64_t& t2) {
-   t1 = IT[0][S[get_byte<7>(x1)]][0] ^ IT[1][S[get_byte<6>(x1)]][0] ^ IT[2][S[get_byte<5>(x1)]][0] ^
-        IT[3][S[get_byte<4>(x1)]][0] ^ IT[4][S[get_byte<3>(x1)]][0] ^ IT[5][S[get_byte<2>(x1)]][0] ^
-        IT[6][S[get_byte<1>(x1)]][0] ^ IT[7][S[get_byte<0>(x1)]][0] ^ IT[8][S[get_byte<7>(x2)]][0] ^
-        IT[9][S[get_byte<6>(x2)]][0] ^ IT[10][S[get_byte<5>(x2)]][0] ^ IT[11][S[get_byte<4>(x2)]][0] ^
-        IT[12][S[get_byte<3>(x2)]][0] ^ IT[13][S[get_byte<2>(x2)]][0] ^ IT[14][S[get_byte<1>(x2)]][0] ^
-        IT[15][S[get_byte<0>(x2)]][0];
-   t2 = IT[0][S[get_byte<7>(x1)]][1] ^ IT[1][S[get_byte<6>(x1)]][1] ^ IT[2][S[get_byte<5>(x1)]][1] ^
-        IT[3][S[get_byte<4>(x1)]][1] ^ IT[4][S[get_byte<3>(x1)]][1] ^ IT[5][S[get_byte<2>(x1)]][1] ^
-        IT[6][S[get_byte<1>(x1)]][1] ^ IT[7][S[get_byte<0>(x1)]][1] ^ IT[8][S[get_byte<7>(x2)]][1] ^
-        IT[9][S[get_byte<6>(x2)]][1] ^ IT[10][S[get_byte<5>(x2)]][1] ^ IT[11][S[get_byte<4>(x2)]][1] ^
-        IT[12][S[get_byte<3>(x2)]][1] ^ IT[13][S[get_byte<2>(x2)]][1] ^ IT[14][S[get_byte<1>(x2)]][1] ^
-        IT[15][S[get_byte<0>(x2)]][1];
+inline void ILS(uint64_t x1, uint64_t x2, uint64_t& t1, uint64_t& t2) {
+   t1 = 0;
+   t2 = 0;
+   for(size_t i = 0; i != 16; ++i) {
+      const uint8_t x = get_byte_var(7 - (i % 8), (i < 8) ? x1 : x2);
+      t1 ^= IT[i][x][0];
+      t2 ^= IT[i][x][1];
+   }
 }
```

Comment:
Indeed, this overload doesn't seem to be used anywhere.

Reply:
Yeah, just leftover. I guess only clang warns about this at the moment. I suppose the compiler has to keep track of which TU contained the definition, since if the inline came from a header the warning would be noise.